File: day-1/JVM/GettingStartedOllama.kt
import java.net.URI
import java.net.http.HttpClient
import java.net.http.HttpRequest
import java.net.http.HttpResponse

fun main() {
    try {
        val client = HttpClient.newHttpClient()

        val jsonString = """{"model":"qwen3","prompt":"Hello","stream":false,"Think":false}"""

        val request = HttpRequest.newBuilder()
            .uri(URI.create("http://localhost:11434/api/generate"))
            .header("Content-Type", "application/json")
            .POST(HttpRequest.BodyPublishers.ofString(jsonString))
            .build()

        val response = client.send(request, HttpResponse.BodyHandlers.ofString())

        val responseBody = response.body()
        val responseText = extractJsonValue(responseBody, "response")
        println(responseText)

    } catch (e: Exception) {
        e.printStackTrace()
    }
}

fun extractJsonValue(json: String, key: String): String {
    val searchKey = "\"$key\":\""
    val startIndex = json.indexOf(searchKey)
    if (startIndex == -1) return ""

    val valueStart = startIndex + searchKey.length
    val endIndex = json.indexOf("\"", valueStart)
    if (endIndex == -1) return ""

    return json.substring(valueStart, endIndex).replace("\\n", "\n").replace("\\\"", "\"")
}

File: day-1/getting_started_ollama.py
import requests
import logging

logging.basicConfig()
logging.getLogger().setLevel(logging.DEBUG)
requests_log = logging.getLogger("urllib3")
requests_log.setLevel(logging.DEBUG)
requests_log.propagate = True

try: # for Python 3
    from http.client import HTTPConnection
except ImportError:
    from httplib import HTTPConnection
HTTPConnection.debuglevel = 1

response = requests.post(
    "http://localhost:11434/api/generate",
    json={"model": "qwen3:4b", "prompt": "Hello",
          "stream": False, "Think": False }
)

data = response.json()
print(data["response"])
